### ISSUE TITLE
[P0] ArmOperationHelpers.UpdateStatusAsync should not throw exception when returning 201

### ISSUE THREAD

**Issue by erich-wang** [Azure/autorest.csharp#743]
This is blocker issue for preview.

For some LROs, 200,201,202 are accepted status code, e.g. [Resources_UpdateById](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json#L3184), however ArmOperationHelpers.UpdateStatusAsync will throw exception when status code is 201.

**Comment by MiYanni** [comment]
@erich-wang 
I spent about 3 weeks researching, investigating, testing, and diagramming ARM LROs. Looking at the spec here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations

I'm assuming you are referring to the final response logic (since we currently accept between 200 to 204 for polling). There is nothing in the spec indicating that `201 Created` is a valid final response status code. However, the information for PUT/PATCH is a bit non-specific in this regard. Can you please provide the specific scenarios where `201 Created` is a valid final response? We would need to add tests for this scenario as our test server does cover this situation.

**Comment by erich-wang** [comment]
@MiYanni 
Thanks for the detail information. We run into this issue when creating generic resource with below request and response, the corresponding record file is [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/resources/Microsoft.Azure.Management.Resource/tests/SessionRecords/LiveResourceTests/CreateResourceWithPlan.json#L69). Please note although this record file is from track 1 but it was recorded 5 months ago and no api version change, so the request and response should almost be same, please let me know if need any other info, thx.

```
    {
      "RequestUri": "/subscriptions/b9f138a1-1d64-4108-8413-9ea3be1c1b2d/resourcegroups/csmrg7955/providers/Sendgrid.Email//accounts/csmr1367?api-version=2015-01-01",
      "EncodedRequestUri": "L3N1YnNjcmlwdGlvbnMvYjlmMTM4YTEtMWQ2NC00MTA4LTg0MTMtOWVhM2JlMWMxYjJkL3Jlc291cmNlZ3JvdXBzL2NzbXJnNzk1NS9wcm92aWRlcnMvU2VuZGdyaWQuRW1haWwvL2FjY291bnRzL2NzbXIxMzY3P2FwaS12ZXJzaW9uPTIwMTUtMDEtMDE=",
      "RequestMethod": "PUT",
      "RequestBody": "{\r\n  \"plan\": {\r\n    \"name\": \"free\",\r\n    \"publisher\": \"Sendgrid\",\r\n    \"product\": \"sendgrid_azure\",\r\n    \"promotionCode\": \"\"\r\n  },\r\n  \"properties\": {\r\n    \"password\": \"p@ss9588\",\r\n    \"acceptMarketingEmails\": false,\r\n    \"email\": \"tiano@email.com\"\r\n  },\r\n  \"location\": \"centralus\",\r\n  \"tags\": {\r\n    \"provision_source\": \"RMS\"\r\n  }\r\n}",
      "RequestHeaders": {
        "x-ms-client-request-id": [
          "3d591769-2eb7-4af6-a98e-dc1426f4eba0"
        ],
        "Accept-Language": [
          "en-US"
        ],
        "User-Agent": [
          "FxVersion/4.6.28207.04",
          "OSName/Windows",
          "OSVersion/Microsoft.Windows.10.0.18363.",
          "Microsoft.Azure.Management.ResourceManager.ResourceManagementClient/3.0.0.0"
        ],
        "Content-Type": [
          "application/json; charset=utf-8"
        ],
        "Content-Length": [
          "333"
        ]
      },
      "ResponseHeaders": {
        "Cache-Control": [
          "must-revalidate, no-cache"
        ],
        "Vary": [
          "Accept-Encoding"
        ],
        "x-ms-correlation-request-id": [
          "b682715a-207f-462f-ab90-a37d9a515385"
        ],
        "x-ms-ratelimit-remaining-subscription-writes": [
          "1198"
        ],
        "X-Ms-Request-Id": [
          "5e0652cf57d108-40487911"
        ],
        "Set-Cookie": [
          "sendgrid_frontend=b5dd9438be13cd3305b0aaf2d16e0276:2927ab2002a4401d5fc5cd23d2b25d640277d848; expires=Sun, 26-Jan-2020 18:51:59 GMT; path=/; httponly"
        ],
        "Server": [
          "nginx"
        ],
        "x-ms-routing-request-id": [
          "WESTUS:20191227T185238Z:b682715a-207f-462f-ab90-a37d9a515385"
        ],
        "Strict-Transport-Security": [
          "max-age=31536000; includeSubDomains"
        ],
        "X-Content-Type-Options": [
          "nosniff"
        ],
        "Date": [
          "Fri, 27 Dec 2019 18:52:37 GMT"
        ],
        "Content-Length": [
          "356"
        ],
        "Content-Type": [
          "application/json"
        ],
        "Expires": [
          "0"
        ]
      },
      "ResponseBody": "{\r\n  \"plan\": {\r\n    \"name\": \"free\",\r\n    \"publisher\": \"Sendgrid\",\r\n    \"product\": \"sendgrid_azure\",\r\n    \"promotionCode\": \"\"\r\n  },\r\n  \"properties\": {\r\n    \"email\": \"tiano@email.com\",\r\n    \"provisioningState\": \"Succeeded\",\r\n    \"status\": \"Running\",\r\n    \"username\": \"azure_146f86bd07af503f05d85352df57cd8c@azure.com\",\r\n    \"smtpServer\": \"smtp.sendgrid.net\",\r\n    \"planLabel\": \"Free\"\r\n  },\r\n  \"location\": \"centralus\",\r\n  \"tags\": {\r\n    \"provision_source\": \"RMS\"\r\n  }\r\n}",
      "StatusCode": 201
    }
```

**Comment by MiYanni** [comment]
I did some more investigation. It seems this is quite a specific scenario to `PUT` only. Looking at the ARM spec I linked to before, for `PUT`:
```
1. Respond to the initial PUT request with a 201 Created or 200 OK (per normal guidance)
```

So, in this situation, the initial request responded with `201 Created`, but it was also 'done polling'; meaning, it had a `"provisioningState": "Succeeded"`. So, this was one specific case that was missed in our generator's test scenarios. I will add this test and add the case statement to allow it in the new LRO code.
